src/IECoreDelight/Renderer.cpp

```diff
@@ -447,6 +452,35 @@ class DelightShader : public IECore::RefCounted
 							0, nullptr
 						);
 					}
+
+					// Add a `uvCoord` shader if needed
+
+					for( const auto &parameter : shader->parameters() )
+					{
+						if( parameter.first == g_uvCoordParameterName )
+						{
+							ShaderNetwork::Parameter uvInput = shaderNetwork->input( ShaderNetwork::Parameter( handle, g_uvCoordParameterName ) );
+
+							if( !uvInput )
+							{
+								NSICreate( context, g_uvCoordNodeName, "shader", 0, nullptr);
```

Comment:
I think we need to create a `DelightHandle` for this and store it in `m_handles`, otherwise we're basically leaking the node, and will never clean it up if the DelightShader gets deleted during IPR. If we do this, we'll need to be careful it doesn't end up at `m_handles.back()`, because that's assumed to be the output shader.

While adding the `uvCoord` shader here is expedient, I'm not sure it's the best place given the future work we intend to do. Support for UsdPreviewSurface and UsdLux will involve more extensive transforms, and it's easier to transform a ShaderNetwork in whatever order you want, than it is to do it all inline here in the order visited by `depthFirstTraverse()`. So I'd suggest we follow the pattern established for Arnold and Cycles, where a separate function transforms the ShaderNetwork _before_ we convert it to 3Delight. This also makes testing easier, because we can query the resulting ShaderNetwork directly, rather than writing an NSI file and doing manual parsing.